livekit/src/room/token_source/fetch_options.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright 2025 LiveKit, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use livekit_protocol::{RoomAgentDispatch, RoomConfiguration, TokenSourceRequest};
+
+/// Options that can be used when fetching new credentials from a TokenSourceConfigurable
+///
+/// Example:
+/// ```rust
+/// let _ = TokenSourceFetchOptions::default().with_agent_name("my agent name");
+/// ```
+#[derive(Clone, Default)]
+pub struct TokenSourceFetchOptions {
+    pub room_name: Option<String>,
```

Comment:
praise: I like using the builder pattern here! I think we should leverage the builder pattern for new APIs since it gives us more room to make changes in the public API without breaking.